hypershift-operator/controllers/hostedcluster/internal/platform/powervs/ccm.go

```diff
@@ -0,0 +1,302 @@
+	image          string
+}
+
+func (ccm setupCCM) getAccountID() (accountID string, err error) {
+	var secret corev1.Secret
+	err = ccm.c.Get(ccm.ctx, client.ObjectKey{Namespace: ccm.hcluster.GetNamespace(), Name: ccm.hcluster.Spec.Platform.PowerVS.ClusterControlManagerCreds.Name}, &secret)
+	if err != nil {
+		err = fmt.Errorf("failed to get control plane operator provider creds %s: %w", ccm.hcluster.Spec.Platform.PowerVS.ClusterControlManagerCreds.Name, err)
+		return
+	}
+	secretData, secretHasData := secret.Data["ibmcloud_api_key"]
+	if !secretHasData {
+		err = fmt.Errorf("cluster control manager credentials secret %q is missing credentials key", secret.Name)
+		return
+	}
+	apiKeyStr := string(secretData)
+	auth := &ibmcore.IamAuthenticator{
+		ApiKey: apiKeyStr,
+	}
+
+	iamv1, err := iamidentityv1.NewIamIdentityV1(&iamidentityv1.IamIdentityV1Options{
+		Authenticator: auth,
+	})
+	if err != nil {
+		return
+	}
+
+	apiKeyDetailsOpt := iamidentityv1.GetAPIKeysDetailsOptions{IamAPIKey: &apiKeyStr}
+	apiKey, _, err := iamv1.GetAPIKeysDetails(&apiKeyDetailsOpt)
+	if err != nil {
+		return
+	}
+	if apiKey == nil {
+		err = fmt.Errorf("could not retrieve account id")
+		return
+	}
+
+	accountID = *apiKey.AccountID
+	return
+}
+
+func (ccm setupCCM) setupConfigMap() (err error) {
+	accountID, err := ccm.getAccountID()
+	if err != nil {
+		err = fmt.Errorf("error retrieving accountID while setting up config map: %w", err)
+		return
+	}
+
+	configMap := configMap{
+		ConfigFile:               fmt.Sprintf("%s/%s", kubeConfigFileBasePath, kubeConfigKeyInSecret),
+		AccountID:                accountID,
+		ClusterID:                ccm.hcluster.Name,
+		G2workerServiceAccountID: accountID,
+		G2Credentials:            fmt.Sprintf("%s/%s", secretMountPath, "ibmcloud_api_key"),
+		G2ResourceGroupName:      ccm.hcluster.Spec.Platform.PowerVS.ResourceGroup,
+		G2VpcSubnetNames:         ccm.hcluster.Spec.Platform.PowerVS.VPC.Subnet,
+		G2VpcName:                ccm.hcluster.Spec.Platform.PowerVS.VPC.Name,
+		Region:                   ccm.hcluster.Spec.Platform.PowerVS.VPC.Region,
+		PowerVSCloudInstanceID:   ccm.hcluster.Spec.Platform.PowerVS.ServiceInstanceID,
+		PowerVSRegion:            ccm.hcluster.Spec.Platform.PowerVS.Region,
+		PowerVSZone:              ccm.hcluster.Spec.Platform.PowerVS.Zone,
+	}
+
+	templateData := `
+[global]
+version = 1.1.0
+[kubernetes]
+config-file = {{.ConfigFile}}
+[provider]
+cluster-default-provider = g2
+accountID = {{.AccountID}}
+clusterID = {{.ClusterID}}
+g2workerServiceAccountID = {{.G2workerServiceAccountID}}
+g2Credentials = {{.G2Credentials}}
+g2ResourceGroupName = {{.G2ResourceGroupName}}
+g2VpcSubnetNames = {{.G2VpcSubnetNames}}
+g2VpcName = {{.G2VpcName}}
+region = {{.Region}}
+powerVSCloudInstanceID = {{.PowerVSCloudInstanceID}}
+powerVSRegion = {{.PowerVSRegion}}
+powerVSZone = {{.PowerVSZone}}`
+
+	configMapTemplate := template.New("configMap")
+	configMapTemplate.Parse(templateData)
```

Comment:
Please do the template parsing in a package-level variable declaration similiar to https://github.com/openshift/hypershift/blob/1d4aca4f1defffc167681e42e45598b6f7bf300a/hypershift-operator/controllers/nodepool/haproxy.go#L154 because:
* If the template parsing fails, we will notice whenever we start the binary and not only when we hit this codepath
* It is more efficient, as we only have to parse once